lib/src/screens/home_screen/home_screen.dart

```diff
@@ -1,16 +1,29 @@
 import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:intl/intl.dart' as intl;
+
+import '/../src/blocs/coin_bloc/coin_bloc.dart';
+import '/../src/blocs/coin_bloc/coin_event.dart';
+import '/../src/blocs/coin_bloc/coin_state.dart';
 
 import '/../src/constants/name_routes_constants.dart';
 import '/../src/constants/string_constants.dart';
 
 import '/../src/widgets/custom_card.dart';
 import '/../src/constants/color_constants.dart';
 
-class HomeScreen extends StatelessWidget {
+class HomeScreen extends StatefulWidget {
   const HomeScreen({Key? key}) : super(key: key);
 
+  @override
+  State<HomeScreen> createState() => _HomeScreenState();
+}
+
+class _HomeScreenState extends State<HomeScreen> {
   @override
   Widget build(BuildContext context) {
+    final percentageFormat = intl.NumberFormat("##0.0#");
+    context.read<CoinBloc>().add(CoinRequested());
```

Comment:
move this line to initState if you do not need to call it multiple times

Reply:
I move it to class Route. And after that I add it after CoinBloc.